onnxruntime/contrib_ops/cpu/bert/attention.cc

```diff
@@ -239,10 +250,39 @@ Status Attention<T>::PrePack(const Tensor& weights, int input_idx, bool& is_pack
     weights_data += head_size;
   }
 
+  if (!kernel_owns_prepacked_buffer) {
```

Comment:
Does the kernel really need to make this difference? The session can call 2 functions: PrePack (given an allocator and index, get the prepacked weights) and then call StorePrePackedWeights to store them in the kernel. The check for whether the kernel read the weights inside Storexxx is good, but other than that the 2 functions should behave the same (with and without cache). We can discuss more offline.

Reply:
Re-iterating what we spoke during the offline sync, this differentiator is needed within the kernel because the kernel sets the buffer deleter to the its pre-packed buffer as nullptr if the kernel doesn't own the buffer and vice-versa.